app/src/main/kotlin/com/simplemobiletools/smsmessenger/extensions/Context.kt

```diff
@@ -713,7 +748,16 @@ fun Context.updateConversationArchivedStatus(threadId: Long, archived: Boolean)
     }
     val selection = "${Threads._ID} = ?"
     val selectionArgs = arrayOf(threadId.toString())
-    contentResolver.update(uri, values, selection, selectionArgs)
+    try {
+        contentResolver.update(uri, values, selection, selectionArgs)
+    } catch (sqliteException: SQLiteException) {
+        if (sqliteException.message?.contains("no such column: archived") == true && config.useThreadsArchive) {
```

Comment:
not sure how reliable is this, we will see

Reply:
I was worried about disabling archive on any kind of `SQLiteException`, especially since that option is hidden to the users, so I tried to make it a bit more specific.